xrpl4j-model/src/main/java/org/xrpl/xrpl4j/model/client/fees/FeeUtils.java

```diff
@@ -0,0 +1,421 @@
+import com.google.common.base.Preconditions;
+import com.google.common.primitives.UnsignedInteger;
+import com.google.common.primitives.UnsignedLong;
+import org.immutables.value.Value.Derived;
+import org.immutables.value.Value.Immutable;
+import org.xrpl.xrpl4j.model.immutables.FluentCompareTo;
+import org.xrpl.xrpl4j.model.ledger.SignerListObject;
+import org.xrpl.xrpl4j.model.transactions.XrpCurrencyAmount;
+
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.math.MathContext;
+import java.math.RoundingMode;
+import java.util.Arrays;
+import java.util.Objects;
+
+/**
+ * Utils relating to XRPL fees.
+ */
+public class FeeUtils {
+
+  private static final BigInteger MAX_UNSIGNED_LONG = UnsignedLong.MAX_VALUE.bigIntegerValue();
+
+  private static final BigDecimal ONE_POINT_ONE = new BigDecimal("1.1");
+
+  private static final BigDecimal ZERO_POINT_ONE = new BigDecimal("0.1");
+
+  private static final BigInteger FIVE_HUNDRED = BigInteger.valueOf(500);
+
+  private static final BigDecimal TWO = new BigDecimal(2);
+
+  private static final BigDecimal THREE = new BigDecimal(3);
+
+  private static final BigInteger FIFTEEN = BigInteger.valueOf(15);
+
+  private static final BigInteger TEN_THOUSAND = BigInteger.valueOf(10000);
+
+  private static final BigInteger ONE_THOUSAND = BigInteger.valueOf(1000);
+
+
+  /**
+   * Computes the fee necessary for a multisigned transaction.
+   *
+   * <p>The transaction cost of a multisigned transaction must be at least {@code (N + 1) * (the normal
+   * transaction cost)}, where {@code N} is the number of signatures provided.
+   *
+   * @param currentLedgerFeeDrops The current ledger fee, represented as an {@link XrpCurrencyAmount}.
+   * @param signerList            The {@link SignerListObject} containing the signers of the transaction.
+   *
+   * @return An {@link XrpCurrencyAmount} representing the multisig fee.
+   */
+  public static XrpCurrencyAmount computeMultiSigFee(final XrpCurrencyAmount currentLedgerFeeDrops,
+    final SignerListObject signerList) {
+    Objects.requireNonNull(currentLedgerFeeDrops);
+    Objects.requireNonNull(signerList);
+
+    return currentLedgerFeeDrops.times(
+      XrpCurrencyAmount.of(UnsignedLong.valueOf(signerList.signerEntries().size() + 1)));
+  }
+
+  /**
+   * Calculate a suggested fee to be used for submitting a transaction to the XRPL. The calculated value depends on the
+   * current size of the job queue as compared to its total capacity.
+   *
+   * @param feeResult {@link FeeResult} object obtained by querying the ledger (e.g., via an `XrplClient#fee()` call).
+   *
+   * @return {@link XrpCurrencyAmount} value of the fee that should be used for the transaction.
+   *
+   * @see "https://xrpl.org/fee.html"
+   * @see "https://github.com/XRPL-Labs/XUMM-App/blob/master/src/services/LedgerService.ts#L244"
+   */
+  public static XrpCurrencyAmount calculateFeeDynamically(final FeeResult feeResult) {
```

Comment:
@mukulljangid @nkramer44 Two questions:

1. I don't like the name of this method because it's unclear what it's doing -- the usage of the word "dynamically" isn't very helpful to me, and it also doesn't really conform to the other public method in this class i.e., `computeMultiSigFee`). Do you have better ideas here? I was thinking maybe just `computeFee` but am curious to hear other opinions here. Xumm calls this `getAvailableNetworkFee`, so we could use `computeAvailableNetworkFee`
2. The Xumm version of this method actually [returns a three-choice object](https://github.com/XRPL-Labs/XUMM-App/blob/8e6e795648b97d0067380baa43bedea9e4b5ea61/src/services/LedgerService.ts#L295) (`low`, `medium` and `high`) and then shows these values to users so they can pick depending on their appetite -- e.g., a Xumm user may not want to always pay the highest fee, and Xumm seems to allow users to select this. I think we should do the same -- anyone using xrpl4j shouldn't be forced into using the fee that we choose for them, because an xrpl4j developer might want to override this behavior. Therefore, I propose that we return an object that contains the low/medium/high fees. WDYT?

Reply:
I like `computeNetworkFees` as well. I agree with point 2 as well. I'll make the change and add javadoc to guide the user.